### PR TITLE
support auto generate for kldiv_loss

### PR DESCRIPTION
**Pull request by enkilee**
### PR types

Others
### PR changes

Others
### Describe

support auto generate for kldiv_loss
- https://github.com/PaddlePaddle/Paddle/issues/51842   
Done: mission 47 kldiv_loss

### CODE REVIEW

File: paddle/phi/api/yaml/backward.yaml [hunk 1]
```diff
@@ -721,6 +721,17 @@
   kernel :
     func : inverse_grad
 
+- backward_op : kldiv_loss_grad
+  forward : kldiv_loss(Tensor x, Tensor label, str reduction) -> Tensor(out)
```
Comment (heavyrain-lzy) [inline paddle/phi/api/yaml/backward.yaml]: 对照`kldiv_loss_op.cc`，`reduction`加上默认值

**Comment by luotao1** [comment]
@heavyrain-lzy 修改完毕，可以再次 review 了

**Comment by heavyrain-lzy** [comment]
由于其他开发者合入了代码现在发生了conflicts,请先解决一下冲突@enkilee

File: paddle/phi/api/yaml/legacy_ops.yaml [hunk 2]
```diff
@@ -862,16 +862,6 @@
   intermediate : saved_mean, saved_variance
   backward : instance_norm_grad
 
-- op : kldiv_loss
-  args : (Tensor x, Tensor label, str reduction)
-  output : Tensor(out)
-  infer_meta :
-    func : KLDivInferMeta
-  kernel :
```
Comment (heavyrain-lzy) [inline paddle/phi/api/yaml/legacy_ops.yaml:861]: 文件冲突，请先解决一下冲突

Comment (enkilee) [reply paddle/phi/api/yaml/legacy_ops.yaml:861]: Done.

File: paddle/phi/api/yaml/legacy_ops.yaml [hunk 3]
```diff
@@ -843,15 +843,17 @@
   intermediate : saved_mean, saved_variance
   backward : instance_norm_grad
 
-- op : kldiv_loss
-  args : (Tensor x, Tensor label, str reduction)
-  output : Tensor(out)
+- op : lamb_
+  args : (Tensor param, Tensor grad, Tensor learning_rate, Tensor moment1, Tensor moment2, Tensor beta1_pow, Tensor beta2_pow, Tensor master_param, Tensor skip_update, float weight_decay, float beta1, float beta2, float epsilon, bool multi_precision)
+  output : Tensor(param_out), Tensor(moment1_out), Tensor(moment2_out), Tensor(beta1_pow_out), Tensor(beta2_pow_out), Tensor(master_param_outs)
```
Comment (heavyrain-lzy) [inline paddle/phi/api/yaml/legacy_ops.yaml]: 这里冲突是不是解错了，develop下的`legacy_ops.yaml`下应该没有`lamb_`这个算子配置了

Comment (enkilee) [reply paddle/phi/api/yaml/legacy_ops.yaml]: sorry ,merge error. already fix it.

**Review by heavyrain-lzy — APPROVED** [review]
LGTM

**Comment by heavyrain-lzy** [comment]
失败的CI请rerun一下@enkilee

**Pull request merged**